### DISCUSSION TITLE
Which the suggested way from Raw[T] to T

### DISCUSSION THREAD

**Discussion by mcallisto** [Iltotore/iron#52] [Q&A] [answered]
I have something like:

```
def unsafe[T](raw: Raw[T]): T =
  raw match
    case Right(t: T) => t
    case _           => throw new Error
```

but I wonder if there is a better approach

**Comment by Iltotore** [comment, marked as answer]
In Iron, `Raw` means `not refined`. In other words, it's a value of type `Constrained` value with an always-true constraint (or "no constraint"). It should always be `Right`.

Your code is actually ok. You can use `getOrElse` to shorten your code but it is mostly a matter of preferences.

Also note that `Raw` will probably disappear in 2.0.0. Seamless interoperability between "normal" types and refined/constrained is targeted:
```scala
val x: Int = ???
val y: Int / AlwaysValidConstraint = x //No error and no verification required in 2.0.0
```
